pkg/controller/autoops/reconcile.go

```diff
@@ -0,0 +1,260 @@
+		state.UpdateWithPhase(autoopsv1alpha1.NoResourcesPhase).
+			UpdateResources(0)
+		return results
+	}
+
+	state.UpdateResources(len(accessibleClusters))
+	readyCount := 0
+	errorCount := 0
+
+	for _, es := range accessibleClusters {
+		log := log.WithValues("es_namespace", es.Namespace, "es_name", es.Name)
+
+		if es.Status.Phase != esv1.ElasticsearchReadyPhase {
+			log.V(1).Info("Skipping ES cluster that is not ready", "es_namespace", es.Namespace, "es_name", es.Name)
+			state.UpdateWithPhase(autoopsv1alpha1.ResourcesNotReadyPhase)
+			results = results.WithRequeue(defaultRequeue)
+			continue
+		}
+
+		if es.Spec.HTTP.TLS.Enabled() {
+			if err := r.reconcileAutoOpsESCASecret(ctx, policy, es); err != nil {
+				log.Error(err, "while reconciling AutoOps ES CA secret", "es_namespace", es.Namespace, "es_name", es.Name)
+				errorCount++
+				state.UpdateWithPhase(autoopsv1alpha1.ErrorPhase)
+				results.WithError(err)
+				continue
+			}
+		}
+
+		apiKeySecret, err := r.reconcileAutoOpsESAPIKey(ctx, policy, es)
+		if err != nil {
+			log.Error(err, "while reconciling AutoOps ES API key", "es_namespace", es.Namespace, "es_name", es.Name)
+			errorCount++
+			state.UpdateWithPhase(autoopsv1alpha1.ErrorPhase)
+			results.WithError(err)
+			continue
+		}
+
+		configMap, err := ReconcileAutoOpsESConfigMap(ctx, r.Client, policy, es)
+		if err != nil {
+			log.Error(err, "while reconciling AutoOps ES config map", "es_namespace", es.Namespace, "es_name", es.Name)
+			errorCount++
+			state.UpdateWithPhase(autoopsv1alpha1.ErrorPhase)
+			results.WithError(err)
+			continue
+		}
+
+		configHash, err := buildConfigHash(ctx, *configMap, *apiKeySecret, r.Client, policy)
+		if err != nil {
+			log.Error(err, "while building config hash", "es_namespace", es.Namespace, "es_name", es.Name)
+			errorCount++
+			state.UpdateWithPhase(autoopsv1alpha1.ErrorPhase)
+			results.WithError(err)
+			continue
+		}
+
+		deploymentParams, err := r.buildDeployment(configHash, policy, es)
+		if err != nil {
+			log.Error(err, "while getting deployment params", "es_namespace", es.Namespace, "es_name", es.Name)
+			errorCount++
+			state.UpdateWithPhase(autoopsv1alpha1.ErrorPhase)
+			results.WithError(err)
+			continue
+		}
+
+		reconciledDeployment, err := deployment.Reconcile(ctx, r.Client, deploymentParams, &policy)
+		if err != nil {
+			log.Error(err, "while reconciling deployment", "es_namespace", es.Namespace, "es_name", es.Name)
+			errorCount++
+			state.UpdateWithPhase(autoopsv1alpha1.ErrorPhase)
+			results.WithError(err)
+			continue
+		}
+
+		if isDeploymentReady(reconciledDeployment) {
+			readyCount++
+		}
+	}
+
+	state.UpdateReady(readyCount).
+		UpdateErrors(errorCount)
+
+	// Schedule a requeue to periodically re-check RBAC permissions.
+	// Use ReconciliationComplete() to indicate this is a periodic check, not an incomplete reconciliation.
+	if rbacResult := requeueRbacCheck(r.accessReviewer); rbacResult.RequeueAfter > 0 {
```

Comment:
Should we reuse `RequeueRbacCheck` from the `association` package?
```suggestion
	if rbacResult := association.RequeueRbacCheck(r.accessReviewer); rbacResult.RequeueAfter > 0 {
```